File: java/src/main/java/com/showmeyourcode/playground/java/code/Lists.java
package com.showmeyourcode.playground.java.code;

import com.showmeyourcode.playground.kotlin.common.Descriptions;

import java.util.List;

import static com.showmeyourcode.playground.java.Application.LOGGER;

public class Lists {

    private Lists(){}

    public static void main(){
        LOGGER.info("\n{} Lists", Descriptions.INDENT1);
        List<Object> mixedTypes = List.of(1,"String",'a',4.2);
        LOGGER.info("mixed types: {}", mixedTypes);
    }
}

File: kotlin/src/main/kotlin/com/showmeyourcode/playground/kotlin/code/Lists.kt
package com.showmeyourcode.playground.kotlin.code

import com.showmeyourcode.playground.kotlin.common.Descriptions
import com.showmeyourcode.playground.kotlin.common.Logging

object Lists {
    fun main() {
        Logging.LOGGER.info("\n{} Lists", Descriptions.INDENT1)
        // mixing types are not recommended
        val mixedTypes: List<Any> =
            mutableListOf(
                "A",
                1,
                2,
                'f'
            )
    }
}
